### ISSUE TITLE
🐛 Bug: @types/node is needed as devDependency for created projects

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [JoshuaKGoldberg/create-typescript-app#1598]
### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/create-typescript-app/issues?q=is%3Aissue) and found none that matched my issue.

### Expected

All devDependencies needed for linting should be installed in new projects.

### Actual

`@types/node` is missed. On new repos:

```plaintext
  52:5   error  Unsafe assignment of an error typed value        @typescript-eslint/no-unsafe-assignment
```

...which is complaining about `import.meta.dirname`. That property comes from `@types/node`.

### Additional Info

💖
